cluster-api/src/main/java/com/sequenceiq/cloudbreak/cluster/service/ClusterComponentConfigProvider.java

```diff
@@ -199,11 +208,26 @@ public void restorePreviousVersion(ClusterComponent clusterComponent) {
         }
     }
 
+    public void cleanUpDetachedEntries() {
+        LOGGER.debug("About to fetch detached {} entries", ClusterComponentHistory.class.getSimpleName());
+        Set<ClusterComponentHistory> detachedEntries = clusterComponentHistoryRepository.findDetachedEntries();
```

Comment:
unnecessary to fetch the all of the cluster component to check the count. Could you please refactor this?